src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -282,6 +282,7 @@ static ParserOps()
         private static readonly string[] s_chars = new string[255];
         internal static readonly object _TrueObject = (object)true;
         internal static readonly object _FalseObject = (object)false;
+        private static readonly CallSite<Func<CallSite, object, bool>> s_toBoolSite = CallSite<Func<CallSite, object, bool>>.Create(PSConvertBinder.Get(typeof(bool)));
```

Comment:
@daxian-dbw It's still a bit unclear to me whether it's advisable to maintain shared callsite caching between the two operators. I basically lifted this technique from `Where-Object` (since we want to mimic that behavior)

Reply:
It's OK to share the callsite as long as the binder associated with it does a very specific type of operations, like in this case.